projects/aca-content/src/lib/components/dl-custom-components/datatable-cell-badges/datatable-cell-badges.component.spec.ts

```diff
@@ -112,4 +100,28 @@ describe('DatatableCellBadgesComponent', () => {
     const dynamicComponent = fixture.debugElement.query(By.css('adf-dynamic-component')).nativeElement;
     expect(dynamicComponent).toBeDefined();
   });
+
+  describe('mouse and keyboard events', () => {
+    let badges: HTMLElement[];
+    let runActionSpy: jasmine.Spy;
+    beforeEach(() => {
+      spyOn(appExtensionService, 'getBadges').and.returnValue(
+        of([{ id: 'test', type: ContentActionType.custom, icon: 'warning', tooltip: 'test tooltip', actions: { click: 'test' } }])
```

Comment:
I would be into define the object in one place, instead three separate almost the same

Reply:
Nice catch, fixed, thanks!